src/fides/api/task/filter_results.py

```diff
@@ -59,14 +59,29 @@ def filter_data_categories(
             itertools.chain(
                 *[
                     field_paths
-                    for cat, field_paths in data_category_fields[
+                    for cat, field_paths in dataset_graph.data_category_field_mapping[
                         CollectionAddress.from_string(node_address)
                     ].items()
                     if any(cat.startswith(tar) for tar in target_categories)
                 ]
             )
         )
 
+        collection_data_categories = set(
+            dataset_graph.nodes[
+                CollectionAddress.from_string(node_address)
+            ].collection.data_categories
+            or []
+        )
+
+        if collection_data_categories:
+            if any(
+                collection_category.startswith(tuple(target_categories))
+                for collection_category in collection_data_categories
+            ):
+                filtered_access_results[node_address].extend(results)
+                continue
+
```

Comment:
The main thing that stands out here is the potential for conflicts between field-level categories and collection-level categories.

For example, my Address id here is specifically `system.operations` - I feel like this indicates this is internal system data that we don't want to hand over to the end user, yet the broad collection-level categories would now cause this to be returned. I think this is surprising behavior.

```yaml
dataset:
  - fides_key: postgres_example_test_dataset
    name: Postgres Example Test Dataset
    description: Example of a Postgres dataset containing a variety of related tables like customers, products, addresses, etc.
    collections:
      - name: address
        data_categories: [user.contact.address]
        fields:
          - name: city
            data_categories: [user.contact.address.city]
          - name: house
            data_categories: [user.contact.address.street]
          - name: id
            data_categories: [system.operations]
            fides_meta:
              primary_key: True
          - name: state
            data_categories: [user.contact.address.state]
          - name: street
            data_categories: [user.contact.address.street]
          - name: zip
            data_categories: [user.contact.address.postal_code]
```


I am wondering if a collection already has field-level categories, should we ignore applying collection-level categories or something along those lines.  

In Fideslang we actually prevent object-level field data categories from being set, instead pushing them to the sub-field level to similarly avoid conflicts, https://github.com/ethyca/fideslang/blob/main/src/fideslang/models.py#L464, although that's not the right solution here.  Similar problem though.

Reply:
I reached out to Neville to get a product-level opinion of how this should work. He decided that what would make sense is for collection-level and field-level data categories to just merge and let the filters apply to this merged set of data categories

Here's an example he provided
```
datasets:
- fides_key: an_extreme_example
  collections:
  - name: users
    data_categories: [user]
    fields:
    - name: accountjson
      data_categories: [user.contact]
      fields:
      - name: accountid
        data_categories: [user.unique_id]
```
the `an_extreme_example.users.accountjson.accountid` field would be treated like it had `[user, user.contact, user.unique_id]`

Reply:
OK. That's not the current behavior correct right, at least with the nested fields, I think Fideslang prevents `data_categories: [user.contact]` from being set on accountjson because it has a nested field.

Reply:
But your work here supports the [user, user.contact] example correct, because you're additive about it, you get the fields data categories first, then layer on the collection categories.  It's just the [user.contact, user.unique_id] merge we don't currently support -

Reply:
You're correct, a more accurate example would be this:
```
datasets:
- fides_key: an_extreme_example
  collections:
  - name: users
    data_categories: [user]
    fields:
    - name: accountjson
      fields:
      - name: accountid
        data_categories: [user.unique_id]
```
We would still prohibit data categories for object-type fields, but the `an_extreme_example.users.accountjson.accountid` field would have a data category of `[user, user.unique_id]`

Reply:
Thanks for confirming!